packages/abi-coder/src/fragments/function-fragment.ts

```diff
@@ -17,16 +18,31 @@ function formatOverride(this: ParamType, format?: string): string {
 export default class FunctionFragment extends Fragment {
   static fromObject(value: JsonFragment): FunctionFragment {
     const { inputs = [], outputs = [] } = value;
+
     const params = {
       type: 'function',
       name: value.name,
-      inputs: inputs.map(ParamType.fromObject),
+      inputs: FunctionFragment.allowOnlyArguments(inputs).map(ParamType.fromObject),
+      strictInputs: FunctionFragment.strictArguments(inputs),
       outputs: outputs.map(ParamType.fromObject),
     };
 
     return new FunctionFragment(params);
   }
 
+  static strictArguments(fragment: readonly JsonFragmentType[]): boolean {
+    return !(
+      fragment.length >= 3 &&
```

Comment:
Note that currently the compiler must emit _exactly_ 4 parameters. It will still work with your code, but if you want to be very specific then this check should be `== 4`. Not a big deal since the compiler will be fixed soon-ish anyways to no longer have this requirement.